src/cluster_legacy.c

```diff
@@ -5387,8 +5396,9 @@ void clusterCron(void) {
         /* The protocol is that function(s) below return non-zero if the node was
          * terminated.
          */
-        if (clusterNodeCronHandleReconnect(node, now)) continue;
+        if (clusterNodeCronHandleReconnect(node, now, &cluster_node_conn_attempts)) continue;
```

Comment:
I still think it would be a bit safer if we initialized at a random position inside the cluster->nodes (a random bucket is fine) and looped until we returned back to the initial node.